### PR TITLE
Integrate Yaksa with MPICH build

### PR DESCRIPTION
**Pull request by gcongiu**
## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

File: src/include/mpir_yaksa.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#ifndef MPIR_YAKSA_H_INCLUDED
+#define MPIR_YAKSA_H_INCLUDED
+
+int MPII_yaksa_init(void);
+int MPII_yaksa_finalize(void);
```
Comment (pavanbalaji) [inline src/include/mpir_yaksa.h:11]: I think these functions can be removed and `yaksa_init/finalize` can be directly called.

File: configure.ac [hunk 2]
```diff
@@ -1210,6 +1210,46 @@ if test "$enable_izem_queue" != "no" && test "$enable_izem_queue" != "none"; the
...
+yaksasrcdir=""
+AC_SUBST([yaksasrcdir])
+yaksalibdir=""
+AC_SUBST([yaksalibdir])
+yaksalib=""
+AC_SUBST([yaksalib])
+
+if test "$with_yaksa_prefix" = "embedded" ; then
+    # no need for libtool versioning when embedding YAKSA
+    yaksa_subdir_args="--disable-versioning --enable-embedded"
```
Comment (pavanbalaji) [inline configure.ac:1236]: We might need to modify yaksa to accept the `--enable-embedding` flag (which can also disable versioning).  Right now it doesn't do that.

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
@pavanbalaji should I close this one?

**Comment by pavanbalaji** [comment]
I thought you were planning to use it as a hack for now.  But if you want to close it, that's fine too.  When I add the typerep integration, I'll have to add this stuff anyway.

**Comment by gcongiu** [comment]
@pavanbalaji yes I can use this but it is very minor. Your work will be doing the true integration so having this PR is probably misleading.

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
@pavanbalaji from CH4 call turned out people would like this to be merged first. Can you review it?

**Comment by gcongiu** [comment]
test:mpich/warnings

**Comment by gcongiu** [comment]
test:mpich/warnings

File: configure.ac (hunk 2 above)
Comment (pavanbalaji) [reply configure.ac:1237]: The `--disable-versioning` flag needs to be removed.  Yaksa would do that when embedding is enabled.

**Comment by gcongiu** [comment]
test:mpich/warnings

**Pull request merged**